src/OSPSuite.Core/Domain/Services/ParameterIdentifications/CategorialParameterIdentificationRunInitializer.cs

```diff
@@ -45,14 +46,22 @@ public override Task<ParameterIdentification> InitializeRun(CancellationToken ca
             MaxDegreeOfParallelism = _coreUserSettings.MaximumNumberOfCoresToUse
          };
 
+         //save which simulations was created based on the old simulation so that we can swap OUTSIDE of the parallel loop
+         var concurrentDictionary = new ConcurrentDictionary<ISimulation, ISimulation>(); 
          return Task.Run(() =>
          {
             var newParameterIdentification = _cloneManager.Clone(ParameterIdentification);
-            Parallel.ForEach(newParameterIdentification.AllSimulations.ToList(), parallelOptions, originalSimulation =>
+            Parallel.ForEach(newParameterIdentification.AllSimulations, parallelOptions, originalSimulation =>
             {
                parallelOptions.CancellationToken.ThrowIfCancellationRequested();
                var newSimulation = createNewSimulationFrom(originalSimulation, _calculationMethodCombination.CalculationMethods);
-               updateReferencesInParameterIdentification(newParameterIdentification, newSimulation, originalSimulation);
+               concurrentDictionary.TryAdd(originalSimulation, newSimulation);
+            });
+
+            concurrentDictionary.Each(kv =>
+            {
+               //Key is the old simulation, value is the corresponding updated new simulation
+               newParameterIdentification.SwapSimulations(kv.Key, kv.Value);
```

Comment:
This was changing the list of simulation in a non thread safe manner. This is now done outside of the Parallel loop